### PR TITLE
Engine/websocket: subscribe to default channels only when actually needed

### PR DESCRIPTION
**Pull request by Rots**
* subscribe if this is required by ws routines or sync manager
* restore previous subscriptions on reconnect (also don't restore anything if there were no subscriptions)

## Type of change
Mostly refactoring
- [x] Bug fix (non-breaking change which fixes an issue)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

start engine with WS disabled
enable WS via gctcli

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Travis with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by shazbert — APPROVED** [review]
tACK, no issues from me.

File: engine/syncer.go [hunk 1]
```diff
@@ -529,9 +529,12 @@ func (e *ExchangeCurrencyPairSyncer) Start() {
 			}
 
 			if !ws.IsConnected() && !ws.IsConnecting() {
-				go WebsocketDataReceiver(ws)
+				go Bot.WebsocketDataReceiver(ws)
 
 				err = ws.Connect()
+				if err == nil {
+					err = ws.FlushChannels()
```
Comment (gloriousCode) [inline engine/syncer.go:536]: Given we have an `else` after `if err != nil {` would you be able to combine the two? Either put both in `if err == nil` or in the `else`

**Review by gloriousCode — CHANGES_REQUESTED** [review]
This is a nice update! Thanks Rots! Just one mini comment

File: engine/syncer.go (hunk 1 above)
Comment (Rots) [reply engine/syncer.go:536]: not quite... I'd still need to add another `err!=nil` check for after the FlushChannels and then duplicate the lines 538-543, I prefer to keep it shorter like this

Comment (gloriousCode) [reply engine/syncer.go:536]: Ahh I missed that. Yeah fair enough.

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for clarifying your reasoning and thank you for another update rots!

**Review by MadCozBadd — APPROVED** [review]
tACK

**Review by thrasher- — APPROVED** [review]

**Pull request merged**